common/src/main/java/com/lightstep/tracer/shared/CollectorClient.java

```diff
@@ -0,0 +1,46 @@
+package com.lightstep.tracer.shared;
+
+import com.lightstep.tracer.grpc.CollectorServiceGrpc;
+import com.lightstep.tracer.grpc.CollectorServiceGrpc.CollectorServiceBlockingStub;
+import com.lightstep.tracer.grpc.ReportRequest;
+import com.lightstep.tracer.grpc.ReportResponse;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import java.util.concurrent.TimeUnit;
+
+public class CollectorClient {
```

Comment:
I'm not clear to me that this abstraction layer is buying us much. It might make more sense if some of the state (counter logic, etc) from AbstractTracer moved in here, though.